environs/config/config.go

```diff
@@ -1515,6 +1523,7 @@ var alwaysOptional = schema.Defaults{
 	CloudInitUserDataKey:          schema.Omit,
 	ContainerInheritPropertiesKey: schema.Omit,
 	BackupDirKey:                  schema.Omit,
+	DefaultSpace:                  schema.Omit,
```

Comment:
@manadart Do we have an upgrade path in mind? Right now I added this to make sure that existing deployments will work.